core/src/main/java/zipkin2/reporter/internal/AsyncReporter.java

```diff
@@ -216,18 +220,15 @@ void startFlusherThread() {
       flushThread.start();
     }
 
+    @SuppressWarnings("unchecked")
     @Override public void report(S next) {
       if (next == null) throw new NullPointerException("span == null");
       // Lazy start so that reporters never used don't spawn threads
       if (started.compareAndSet(false, true)) startFlusherThread();
       metrics.incrementSpans(1);
-      int nextSizeInBytes = encoder.sizeInBytes(next);
-      int messageSizeOfNextSpan = sender.messageSizeInBytes(nextSizeInBytes);
-      metrics.incrementSpanBytes(nextSizeInBytes);
```

Comment:
not sure we have tests for this, but make sure that in countboundedqueue, this increments as soon as we know the size (likely when we are checking it isn't > maxMessageBytes)

Reply:
we could bring the change back , one sec